src/components/CourseItem.jsx

```diff
@@ -1,31 +1,114 @@
-          Edit
-        </Button>
-        <Button variant="danger" className="ml-2">
-          Delete
-        </Button>
-      </Card.Body>
-    </Card>
-  );
+
+function CourseItem({course}) {
+    const navigate = useNavigate();
+    const [showDetails, setShowDetails] = useState(false);
+    const [showDeleteModal, setShowDeleteModal] = useState(false);
+    const {subjectID} = useContext(UserContext);
+
+    const handleEditClick = () => {
+        navigate(`/admin/edit-course?id=${course.courseId}`);
+    };
+
+    const handleDeleteClick = () => {
+        setShowDeleteModal(true);
+    };
+
+    const handleConfirmDelete = () => {
+        api
+            .delete(`/admin/courses/${course.courseId}`,
+                {
+                    params: {
+                        sub: subjectID
+                    }
+                })
+            .then(() => {
+                setShowDeleteModal(false);
+                toast.success("Successfully deleted Course :" + course.courseName)
+            })
+            .catch(() => {
+                toast.error("Something Went Wrong. Please try Again later.")
+            })
+    };
+
+    const handleCancelDelete = () => {
+        setShowDeleteModal(false);
+    }
+
+
+    return (
+        <Card className="mb-3 course-item-card card-container">
+            <Card.Body>
+                <div className="d-flex align-items-center justify-content-between">
+                    <Card.Title>{course.courseName}</Card.Title>
+                    <Button variant="link" onClick={() => setShowDetails((prev) => !prev)}>
+                        {showDetails ? (
+                            <>
+                                Show Less <FiChevronUp/> {/* Show icon for Show Less */}
+                            </>
+                        ) : (
+                            <>
+                                Show More <FiChevronDown/> {/* Show icon for Show More */}
+                            </>
+                        )}
+                    </Button>
+                </div>
+                <Card.Text>Credits: {course.credits}</Card.Text>
+                <Card.Text>Difficulty (1-10): {course.difficultyLevel}</Card.Text>
+
+                {/* Additional details collapse */}
+                <Collapse in={showDetails}>
+                    <div>
+                        <Card.Text>Recommended Study Time (Days): {course.recommendedStudyTime}</Card.Text>
+                        <div>
+                            <Card.Text>Subjects:</Card.Text>
+                            <ul style={{listStyleType: "none", margin: 0, padding: 0}}>
+                                {course.courseSubjects.map((subject, index) => (
+                                    <li key={index}>{`${index + 1}. ${subject}`}</li>
+                                ))}
+                            </ul>
+                        </div>
+                    </div>
+                </Collapse>
```

Comment:
@Danielsio I think you should also add the "Subjects Practice Percentage" field that we have in a course's info.

Reply:
done